.clang-format

```diff
@@ -0,0 +1,46 @@
+# .clang-format
+
+# Base style to use
+BasedOnStyle: Google
+
+# Indentation settings
+IndentWidth: 4
+UseTab: Never
+
+# Alignment settings
+AlignConsecutiveDeclarations: true
+AlignConsecutiveAssignments: true
+AlignTrailingComments: true
+
+# Break settings
+BreakBeforeBraces: Custom
+BraceWrapping:
+  AfterEnum: true
+  AfterStruct: false
+  SplitEmptyFunction: false
+  AfterFunction: true
+  BeforeElse: true
+
+AllowShortBlocksOnASingleLine: false
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: Inline
+AllowShortIfStatementsOnASingleLine: false
+AllowShortLoopsOnASingleLine: false
+
+# Space settings
+SpaceBeforeParens: Custom
+SpaceBeforeParensOptions:
+  AfterControlStatements: false
+  AfterFunctionDefinitionName: false
+
+SpaceAfterCStyleCast: true
+#SpaceBeforeCStyleCast: true
```

Comment:
Do not comment line, just switch to false if it required.

Reply:
I found that the SpaceBeforeCStyleCast is not supported in clang-format
```
Error reading ../.clang-format: Invalid argument
../.clang-format:37:1: error: unknown key 'SpaceBeforeCStyleCast'
SpaceBeforeCStyleCast: false
^~~~~~~~~~~~~~~~~~~~~
```